### PR TITLE
fix "Unknown content type null" in logs and since for 1.12.0 api

### PR DESCRIPTION
**Pull request by lanwen**
+ some small changes (see commits) to help implementing future prs

[Review on Reviewable](https://reviewable.io/reviews/jenkinsci/github-plugin/66)

### CODE REVIEW

File: src/main/java/com/cloudbees/jenkins/GitHubRepositoryName.java [hunk 1]
```diff
@@ -118,6 +118,8 @@ protected GHRepository adapt(GitHub item) {
      *
      * This is useful if the caller only relies on the read access to the repository and doesn't need to
      * walk possible candidates.
+     *
+     * Result can be null
      */
     public GHRepository resolveOne() {
```
Comment (KostyaSha) [inline src/main/java/com/cloudbees/jenkins/GitHubRepositoryName.java]: annotate with CheckForNull

File: src/main/java/org/jenkinsci/plugins/github/util/misc/NullSafeFunction.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.google.common.base.Function;
+import com.google.common.base.Preconditions;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * This abstract class calls {@link #applyNullSafe(Object)} only after success validation of inner object for null
+ * 
+ * {@inheritDoc}
```
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/github/util/misc/NullSafeFunction.java]: shouldn't parameters be inherited by default without `inheritDoc`?

File: src/main/java/org/jenkinsci/plugins/github/util/misc/NullSafePredicate.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ *
+ * {@inheritDoc}
+ *
+ * @author lanwen (Merkushev Kirill)
+ */
+
+public abstract class NullSafePredicate<T> implements Predicate<T> {
+
+    /**
+     * {@inheritDoc}
```
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/github/util/misc/NullSafePredicate.java]: afaik they will be inherited by default, just remove

File: src/main/java/org/jenkinsci/plugins/github/webhook/GHEventPayload.java [hunk 4]
```diff
@@ -80,7 +87,8 @@ public String apply(StaplerRequest request) {
         }
 
         /**
-         * used for application/json content-type 
+         * used for application/json content-type
```
Comment (KostyaSha) [inline src/main/java/org/jenkinsci/plugins/github/webhook/GHEventPayload.java]: pure description...

Comment (lanwen) [inline src/main/java/org/jenkinsci/plugins/github/webhook/GHEventPayload.java]: on next line - `@return function to extract payload from body` think its enough

File: src/main/java/com/cloudbees/jenkins/GitHubRepositoryName.java [hunk 5]
```diff
@@ -118,7 +119,10 @@ protected GHRepository adapt(GitHub item) {
      *
      * This is useful if the caller only relies on the read access to the repository and doesn't need to
      * walk possible candidates.
+     *
+     * Result can be null
```
Comment (KostyaSha) [inline src/main/java/com/cloudbees/jenkins/GitHubRepositoryName.java]: remove this line, checkfornull is enough or if you want java doc then change to `@return desc`

**Pull request merged**